python/ray/tune/trial_runner.py

```diff
@@ -807,16 +807,23 @@ def _validate_result_metrics(self, result):
                                             or "done" not in result):
             base_metric = self._metric
             scheduler_metric = self._scheduler_alg.metric
-            search_metric = self._search_alg.metric
+            search_metrics = self._search_alg.metric
+            if isinstance(search_metrics, str):
+                search_metrics = [search_metrics]
 
             if base_metric and base_metric not in result:
                 report_metric = base_metric
                 location = "tune.run()"
             elif scheduler_metric and scheduler_metric not in result:
                 report_metric = scheduler_metric
                 location = type(self._scheduler_alg).__name__
-            elif search_metric and search_metric not in result:
-                report_metric = search_metric
+            elif search_metrics and any([
+                    search_metric not in result
+                    for search_metric in search_metrics
+            ]):
+                report_metric = list(
+                    filter(lambda search_metric: search_metric not in result,
+                           search_metrics))
```

Comment:
```suggestion
                           search_metrics))
                   if len(report_metric) == 1:
                       report_metric = report_metric[0]
```

Reply:
(please check for correct indentation and linting)

Reply:
LGTM :+1: I added this (and linted it)